front/lib/api/assistant/generation.ts

```diff
@@ -226,30 +196,39 @@ export async function renderConversationForModelMultiActions({
       // Allow at least tokensMargin tokens in addition to the truncation message.
       tokensUsed + approxTruncMsgTokenCount + tokensMargin < allowedTokenCount
     ) {
-      const msg = messages[i] as ContentFragmentMessageTypeModel;
       const remainingTokens =
         allowedTokenCount - tokensUsed - approxTruncMsgTokenCount;
 
       const updatedContent = [];
-      for (const c of msg.content) {
+      for (const c of currentMessage.content) {
         if (!isTextContent(c)) {
           // If there is not enough room and it's an image, we simply ignore it.
           continue;
         }
 
-        const contentRes = await tokenSplit(c.text, model, remainingTokens);
+        // Remove only if it ends with "</attachment>".
+        const textWithoutClosingAttachmentTag = c.text.replace(
```

Comment:
surprised you didn't  create a function tokenSplitAttachment here :p

Reply:
Don't worry I'll take a stab at the generation and eventually create this function 😄